### PR TITLE
Tags layout

### PR DESCRIPTION
**Pull request by madeuz**
closes CRJI/EIC#589

### CODE REVIEW

**Comment by madeuz** [comment]
CRJI/EIC#589

File: src/components/document/Document.js [hunk 1]
```diff
@@ -194,60 +300,125 @@ function Document({ docUrl, data, loading, fullPage, showToolbar = true, showMet
...
+                    <TabStyle icon={<AccountTreeOutlined />} label="Headers &amp; Parts" classes={tabClasses} />
+                )}
 
-            <TextSection
-                title="Headers &amp; Parts"
-                text={data.content.tree}
-                fullPage={fullPage}
-            />
+                {ocrData.map(({tag}) => (
+                    <TabStyle icon={<TextFields />} label={"OCR " + tag} classes={tabClasses} key={tag} />
```
Comment (gabriel-v) [inline src/components/document/Document.js:367]: Sorry, I haven't specified initially, but the OCR texts need to go as sub-tabs of the Text tab.


Having separate tabs for "Text", "OCR TESSERACT_ENG", "OCR_TESSERACT_FRA" and "OCR CUSTOM 1" will take up all the space available for tabs.


The `text` field and the `ocrtext.$ID` are all different variants of the same text, so we'll be running NLP independently for each of these in the future. That's why I think we would want them tabbed too.


For the implicit text tab, show it under a sub-tab with the title "Extracted from file", so we'll have all texts on the same level.

File: src/components/document/Document.js [hunk 2]
```diff
@@ -194,60 +300,125 @@ function Document({ docUrl, data, loading, fullPage, showToolbar = true, showMet
...
 
             {ocrData.map(({tag, text}) => (
-                <TextSection
-                    key={tag}
-                    title={"OCR " + tag}
-                    text={text}
-                    fullPage={fullPage}
-                    omitIfEmpty={false}
-                />
+                <TabPanel value={tab} index={tabIndex++} key={tag}>
```
Comment (gabriel-v) [inline src/components/document/Document.js:434]: what's up with the weird duplication of all these fields here and on line 339?

We want to be moving towards field definitions that come from the backend, not repeated hand-rolled loops...

Maybe some OOP / abstraction would help here?

File: src/components/search/ResultItem.js [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+                                <Tooltip title="Open in new tab">
+                                    <IconButton size="small" style={{ marginRight: 15 }}>
+                                        <a href={url} target="_blank" className={classes.buttonLink}>
+                                            <Launch className={classes.actionIcon} />
+                                        </a>
+                                    </IconButton>
+                                </Tooltip>
+                            </Grid>
+
+                            {fields[`priv-tags.${whoAmI.username}`]?.includes('important') && (
```
Comment (gabriel-v) [inline src/components/search/ResultItem.js:181]: same as above: please get the icon information from some list with tags configuration, so we don't have to edit this code here every time we want to change tag name / color / tooltips / fieldname, or if we want to make these configurable...

File: src/utils.js [hunk 4]
```diff
@@ -27,6 +27,20 @@ export const getIconImageElement = fileType => {
     return img
 }
 
+export const getIconReactComponent = fileType => {
```
Comment (gabriel-v) [inline src/utils.js:30]: so mabybe another one of these, for the tags-related icon objects?

**Review by gabriel-v — CHANGES_REQUESTED** [review]
"important" vs "interesting" gives me dyslexia 

- let's call the yellow private tag just "starred" - sorta like before, but works better with "recommended" and "seen" which are all adjectives 
- let's shamelessly rename green public tag into "recommended" exactly like datashare - that's the one that makes most sense to me

**Comment by gabriel-v** [comment]
Also, some differenced from the issue in the layout:

- please add fillename as document title between the header bar (the one with the star icons and links) and the tabs list (like datashare)
- please add collection name as subtitle under main title mentioned above
- please duplicate all tags from document, taken from Tags API, under collection name subtitle mentioned above

These 3, then the tabs with "Text", "Preview" and others.

Same format for title, subtitle and tags listed should appear on all pages: `_directory_1`, `_file_1` and hash, with or without print mode.

You can keep the [smaller title above the Finder "File /eml-9-pgp/encrypted-machine-learning-comes.eml.asc"](https://hoover.fuze.vîj.ro/doc/testdata/_file_20) unchanged for now - we'll revisit the Finder later.

**Comment by gabriel-v** [comment]
Also, please rename the buttons for the tags name actions like in the issue:

- mark as trash / mark seen
- unmark as trash, unmark seen

Also, please make the "trash" and "seen" tags public instead of private.

Also please change the tooltips for starred and recommended (gold and green) to:
- mark/unmark as starred (private, for yourself)
- mark/unmark as recommended (public, for everybody)
- mark / unmark trash (public)
- mark / unmark seen (public)

Also, please always show the

File: src/components/document/Email.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    const printMode = isPrintMode()
+    const to = (doc.content.to || []).filter(Boolean).join(', ')
+
+    return (
+        <Table>
+            <TableBody>
+                <TableRow>
+                    <TableCell>
+                        {doc.content.from?.length && !printMode ?
+                            <Link href={searchPath(doc.content.from, SEARCH_FROM, collection)} shallow>
```
[the remark this answers is not in the document]
Comment (madeuz) [reply src/components/document/Email.js:34]: to do in CRJI/EIC#618

**Comment by gabriel-v** [comment]
One more layout change:

- on the search results card, instead of highlighting with pink just the left margin of selected document, highlight with pink all the margins 

And:
in the text tab, above the sub-tabs with different text sources, please show email meta fields that dissapeared off into separate "Email" tab

File: src/components/search/ResultItem.js (hunk 3 above)
Comment (madeuz) [reply src/components/search/ResultItem.js]: configuration added

File: src/utils.js (hunk 4 above)
Comment (madeuz) [reply src/utils.js:30]: added tags configuration

File: src/components/document/Document.js (hunk 2 above)
Comment (madeuz) [reply src/components/document/Document.js:396]: added configuration

File: src/components/document/Document.js [hunk 6]
```diff
@@ -92,100 +149,149 @@ function Document({ docUrl, data, loading, fullPage, showToolbar = true, showMet
...
-            href: docUrl,
-            text: 'Open in new tab',
-            icon: <Launch />,
-            target: '_blank',
-        });
-    }
-
     if (data.content.filetype !== 'folder') {
         const important = tags.find(tag => tag.tag === 'important')
         headerLinks.push({
```
[the remark this answers is not in the document]
Comment (madeuz) [reply src/components/document/Document.js]: configuration added

**Review by gabriel-v — CHANGES_REQUESTED** [review]
- [x] group the icons in the document header (tags left / navigation center / others right)
- [x] in search results always reset tab and sub-tab to first
- [x] when OCR documents are present (search `ocr:true`) keep the same style: show the TT icon above the "OCR TESSERACT_RON" sub-tab
- [x] when multiple text sub-tabs exist, instead of picking the first one (which is usually empty or very short), we should reset the sub-tab to the second one (e.g. in screenshot below, select the OCR TESSERACT_ENG tab when navigating to the Text tab)
- [ ] when some person adds a public tag, you should still be able to add yours. 
  -  show a counter in the top bar for the number of public tags of that type set  by everybody
  -  only show the icon colored if you have your public tag set too
-  [x] change seen and trash icons to appear only in top bar, not in tags section
  - keep feature in the code for now, we'll be adding custom tag buttons per collection shortly
-  [ ] change default functionality when adding a tag in input and hitting ENTER: create tag as public, not private

**Comment by gabriel-v** [comment]
- [ ] add tooltips in tags display under document title
  - it's probably useful to see when and who tagged this without looking at the tags editor tab at all
  - show the same tooltip as before
  - also show `Tagged by user <i>username</i>.` or `Tagged by you.` in last line of tooltip for the title-bar tooltips
- [ ] reserve space under title for tags, even if no tags are set
  - when adding/removing one tag, the whole view jumps up and down. That's not nice.

File: src/components/search/filters/Filters.js [hunk 7]
```diff
@@ -96,6 +96,30 @@ function Filters({ loading, query, aggregations, applyFilter, ...rest }) {
                 emptyDisabled
                 {...filterProps}
             />
+
+            <TermsAggregationFilter
+                title="From"
```
Comment (gabriel-v) [inline src/components/search/filters/Filters.js:101]: Rename title to "Email From" to match previous one

File: src/components/search/filters/Filters.js [hunk 8]
```diff
@@ -96,6 +96,30 @@ function Filters({ loading, query, aggregations, applyFilter, ...rest }) {
...
+                field="from.keyword"
+                onLoadMore={handleLoadMore}
+                emptyDisabled
+                {...filterProps}
+            />
+
+            <TermsAggregationFilter
+                title="To"
```
Comment (gabriel-v) [inline src/components/search/filters/Filters.js:109]: Rename title to "Email To" to match previous one

File: src/components/search/filters/Filters.js [hunk 9]
```diff
@@ -96,6 +96,30 @@ function Filters({ loading, query, aggregations, applyFilter, ...rest }) {
...
+                field="to.keyword"
+                onLoadMore={handleLoadMore}
+                emptyDisabled
+                {...filterProps}
+            />
+
+            <TermsAggregationFilter
+                title="Path parts"
```
Comment (gabriel-v) [inline src/components/search/filters/Filters.js:117]: Rename title to just "Path"


I know the archives appear twice in the bucket list (with and without / at end), i'll remove them from the backend some other time.

**Review by gabriel-v — APPROVED** [review]
Really cool! I want those buckets renamed and then just this:



this does seem like a little too much space wasted, right?




let's eliminate some of that space above and below the chips here - or maybe we try something else here, like putting the tags aligned right on the same level as the collection name subtitle?

Cut some space and merge it into master

**Pull request merged**